hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -1375,33 +1375,15 @@ public static Set<String> getValidInstantTimestamps(HoodieTableMetaClient dataMe
   /**
    * Checks if the Instant is a delta commit and has a valid suffix for operations on MDT.
    *
+   * @param datasetPendingInstants The dataset pending instants
    * @param instant {@code HoodieInstant} to check.
    * @return {@code true} if the instant is valid.
    */
-  public static boolean isValidInstant(HoodieInstant instant) {
-    // Should be a deltacommit
-    if (!instant.getAction().equals(HoodieTimeline.DELTA_COMMIT_ACTION)) {
-      return false;
-    }
-
-    // Check correct length. The timestamp should have a suffix over the timeline's timestamp format.
-    final String instantTime = instant.getTimestamp();
-    if (!(instantTime.length() == MILLIS_INSTANT_ID_LENGTH + OperationSuffix.METADATA_INDEXER.getSuffix().length())) {
-      return false;
```

Comment:
Should this be kept?

Reply:
Remove it.

Reply:
hey. can you point me to the code for this 
```
currently only the index init instant has a suffix which might be comprised by last data set completed instant + %3d or SOLO_COMMIT_TIMESTAMP + %3d.
```